php/WP_CLI/Configurator.php

```diff
@@ -80,6 +80,22 @@ public function __construct( $path ) {
 		}
 	}
 
+	/**
+	 * Loads the config spec file.
+	 *
+	 * @param string $path Path to the config spec file.
+	 */
+	private function load_config_spec( $path ) {
+		$config_spec = include $path;
+		// A way for platforms to modify $config_spec.
+		// Use with caution!
+		if ( getenv( 'WP_CLI_INCLUDE_AFTER_CONFIG_SPEC_LOAD' )
+			&& is_readable( getenv( 'WP_CLI_INCLUDE_AFTER_CONFIG_SPEC_LOAD' ) ) ) {
+			include getenv( 'WP_CLI_INCLUDE_AFTER_CONFIG_SPEC_LOAD' );
+		}
```

Comment:
I think it would be better if this was a filter.
The platform will be responsible for making this available within PHP, for example with a WP-CLI package or a MU-plugin.
```suggestion
		$config_spec_filter_callback = getenv( 'WP_CLI_CONFIG_SPEC_FILTER_CALLBACK' );
		if ( callable( $config_spec_filter_callback ) ) {
			$config_spec = $config_spec_filter_callback( $config_spec );
		}
```

Reply:
> I think it would be better if this was a filter.
> The platform will be responsible for making this available within PHP, for example with a WP-CLI package or a MU-plugin.

@schlessera `$spec` is only defined once, and WP-CLI packages and mu-plugins are loaded much later than `$spec` is defined.

As far as I can tell, the only way to inject a callable is with `auto_prepend_file` ([ref](https://www.php.net/manual/en/ini.core.php#ini.auto-prepend-file)). This seems a bit complex, though.

Thoughts?